### PR TITLE
[1LP][RFR] Automating RFE for open url general objects

### PR DESCRIPTION
**Pull request by digitronik**
## Purpose or Intent
- One step toward automating Open Url support for `User`, `Group` and `Tenanat`
- RFE: https://bugzilla.redhat.com/show_bug.cgi?id=1550002
- **one test will fail for tenant**

**Note:** RFE is in progress means Faild ON_QA but still I would like to make it as part of integration test as some objects are working fine.

### PRT Run
{{pytest: cfme/tests/automate/custom_button/test_generic_objects.py::test_custom_button_open_url_evm_obj -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 944a9619a4468d6c4a8adbd1e8c12e355cfd7b59:

`cfme/tests/automate/custom_button/test_generic_objects.py`:
- :bangbang: Line 340:45: **E231** *missing whitespace after ','*
- :bangbang: Line 352:45: **E231** *missing whitespace after ','*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9452707219e8a1d76db9e339efb69864cc7f7ea7

The *local* fixture **`method`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_objects.py**
    - *test_custom_button_open_url_evm_obj*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/custom_button/test_generic_objects.py [hunk 1]
```diff
@@ -55,6 +56,22 @@ def setup_obj(appliance, button_group):
     return obj
 
 
+@pytest.fixture(scope="module")
+def method(custom_instance, button_group):
+    _, obj_type = button_group
+    obj_type = f"miq_{obj_type.lower()}" if obj_type == "GROUP" else obj_type.lower()
```
Comment (jaryn) [inline cfme/tests/automate/custom_button/test_generic_objects.py]: it doesn't seem right to redefine the `obj_type` here

Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: I'm redefine it as per script demands. can you please explain me exact problem?

Comment (jaryn) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: Aha sorry. I didn't see that the object_type is in the RHS of the `=`.

File: cfme/tests/automate/custom_button/test_generic_objects.py [hunk 2]
```diff
@@ -345,4 +361,39 @@ def test_custom_button_open_url_evm_obj(setup_obj, button_group):
...
+        text=fauxfactory.gen_alphanumeric(start="btn", separator="_"),
+        hover=fauxfactory.gen_alphanumeric(),
+        open_url=True,
+        system="Request",
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
```
Comment (john-dupuy) [inline cfme/tests/automate/custom_button/test_generic_objects.py]: @digitronik Why is this `Dropdown` widget not part of `setup_obj`'s view?

File: cfme/tests/automate/custom_button/test_generic_objects.py [hunk 3]
```diff
@@ -345,4 +361,39 @@ def test_custom_button_open_url_evm_obj(setup_obj, button_group):
...
+    assert custom_button_group.has_item(button.text)
+
+    # TODO: Move windows handling functionality to browser
+    initial_count = len(view.browser.selenium.window_handles)
+    main_window = view.browser.selenium.current_window_handle
+    custom_button_group.item_select(button.text)
+
+    wait_for(
+        lambda: len(view.browser.selenium.window_handles) > initial_count,
+        timeout=120,
```
Comment (john-dupuy) [inline cfme/tests/automate/custom_button/test_generic_objects.py]: 2 min to wait for a tab to open seems a bit long -- can this be shortened at all?

File: cfme/tests/automate/custom_button/test_generic_objects.py [hunk 4]
```diff
@@ -345,4 +361,39 @@ def test_custom_button_open_url_evm_obj(setup_obj, button_group):
     Bugzilla:
         1550002
     """
-    pass
+    group, obj_type = button_group
+
+    button = group.buttons.create(
```
Comment (john-dupuy) [inline cfme/tests/automate/custom_button/test_generic_objects.py]: Q: thoughts on moving this into a fixture?

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 2 above)
Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: We already discussed about this @john-dupuy 
https://github.com/ManageIQ/integration_tests/pull/9020#discussion_r299924973

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 4 above)
Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: @john-dupuy our test is related to `open_url`. we are enabling `open url` while creating button [L369].
So as per automation practice; whatever, we want to test better to make part of test `steps`.

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 3 above)
Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: Its maximum time... as we are using it in `wait_for`; If condition satisfy it will take like `10 sec`. can i reduce it to 60s?

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 2 above)
Comment (john-dupuy) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: Ah, I knew that looked familiar :grin:

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 3 above)
Comment (john-dupuy) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: 60s would be better, my only concern with using large timeouts is in the case of a failure. Why wait 2 min when you know after 30 sec that condition has failed? This can potentially add time to our test runs.

Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: ack :). after all timeout is decided by try and error...

**Comment by john-dupuy** [comment]
@digitronik can you check 5.11 failures?

File: cfme/tests/automate/custom_button/test_generic_objects.py [hunk 5]
```diff
@@ -345,4 +361,39 @@ def test_custom_button_open_url_evm_obj(setup_obj, button_group):
...
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+
+    # TODO: Move windows handling functionality to browser
+    initial_count = len(view.browser.selenium.window_handles)
```
Comment (jawatts) [inline cfme/tests/automate/custom_button/test_generic_objects.py]: **Required** We shouldn't have selenium calls in a test case, this should be moved to the framework. I think that move should be part of this PR. Or this PR will be dependent on another PR addresses the framework enhancement.

Comment (digitronik) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: Hey, 
I was planning to do this latter as not that much necessary so added `TODO`. let handle this :+1: I wrapped selenium important method in `wt.core` Browser; so other team also take benefit of this.

Now this PR depends on: https://github.com/RedHatQE/widgetastic.core/pull/157

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 75b2a17acea042b59b4a1e569fbf6aae48bd15a2

The *local* fixture **`method`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_objects.py**
    - *test_custom_button_open_url_evm_obj*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/custom_button/test_generic_objects.py (hunk 5 above)
Comment (izapolsk) [reply cfme/tests/automate/custom_button/test_generic_objects.py]: @jawatts, do you mind if I merge it as is ?
The thing is that managing windows isn't implemented in widgetastic yet and requires some discussion.
From my point, widgetastic should have Window class instead of wrapping selenium calls.

**Comment by izapolsk** [comment]
@digitronik, please update your PR with new wt.core
